c7n/mu.py

```diff
@@ -52,28 +49,46 @@
 
 
 class PythonPackageArchive(object):
-    """Creates a zip file for python lambda functions
+    """Creates a zip file for python lambda functions.
 
-    Packages up a virtualenv and a source package directory per lambda's
-    directory structure.
+    :param tuple modules: the Python modules to add to the archive
 
-    See http://docs.aws.amazon.com/lambda/latest/dg/with-s3-example-deployment-pkg.html#with-s3-example-deployment-pkg-python
-    """
+    Amazon doesn't give us straightforward docs here, only `an example
+    <http://docs.aws.amazon.com/lambda/latest/dg/with-s3-example-deployment-pkg.html#with-s3-example-deployment-pkg-python>`_,
+    from which we can infer that they simply unzip the file into a directory on
+    ``sys.path``. So what we do is locate all of the ``modules`` specified, and
+    add all of the ``.py`` files we find for these modules to a zip file.
 
-    def __init__(self, src_path, virtualenv_dir=None, skip=None,
-                 lib_filter=None, src_filter=None):
+    In addition to the modules specified during instantiation, you can add
+    arbitrary additional files to the archive using :py:func:`add_file` and
+    :py:func:`add_contents`. For example, since we only add ``*.py`` files for
+    you, you'll need to manually add files for any compiled extension modules
+    that your Lambda requires.
+
+    """
 
-        self.src_path = src_path
-        if virtualenv_dir is None:
-            virtualenv_dir = os.path.abspath(
-                os.path.join(os.path.dirname(sys.executable), '..'))
-        self.virtualenv_dir = virtualenv_dir
-        self._temp_archive_file = None
-        self._zip_file = None
+    def __init__(self, *modules):
+        self._temp_archive_file = tempfile.NamedTemporaryFile()
+        self._zip_file = zipfile.ZipFile(
+            self._temp_archive_file, mode='w',
+            compression=zipfile.ZIP_DEFLATED)
         self._closed = False
-        self.lib_filter = lib_filter
-        self.src_filter = src_filter
-        self.skip = skip
+
+        for module in modules:
+            path = imp.find_module(module)[1]
+            if os.path.isfile(path):
+                if not path.endswith('.py'):
```

Comment:
fwiw we don't technically a .py source only, we can do pyc and .so (only if compiled on amz linux). the .py thing was a choice to avoid getting duplicate copies, there's some minor startup time improvements to using pyc

also feels like this logic could be extracted to a separate method